lib/trino-collect/src/main/java/io/trino/collect/cache/CacheUtils.java

```diff
@@ -29,7 +29,7 @@ public static <K, V> V uncheckedCacheGet(Cache<K, V> cache, K key, Supplier<V> l
         }
         catch (ExecutionException e) {
             // this can not happen because a supplier can not throw a checked exception
-            throw new RuntimeException("Unexpected checked exception from cache load", e);
+            throw new RuntimeException("Unexpected checked exception from cache load", e.getCause());
```

Comment:
Because of `// this can not happen because a supplier can not throw a checked exception`  comment, I don't think we need to change this line.

Reply:
This change makes `uncheckedCacheGet` with loader compatible with how `com.google.common.cache.LocalCache.LocalLoadingCache#getUnchecked` works:
```
    @Override
    public V getUnchecked(K key) {
      try {
        return get(key);
      } catch (ExecutionException e) {
        throw new UncheckedExecutionException(e.getCause());
      }
    }
```

There is no need to wrap actual exception (which can happen and can be `TrinoException`) in `ExecutionException`

Reply:
> which can happen to be TrinoException

we're wrapping in RTE anyway, so that's not enough to cater for TrinoException

> This change makes uncheckedCacheGet with loader compatible with how com.google.common.cache.LocalCache.LocalLoadingCache#getUnchecked works

so is the comment incorrect?

the javadoc of the method being called says:

```
Throws:
  ExecutionException – if a checked exception was thrown while loading the value
  UncheckedExecutionException – if an unchecked exception was thrown while loading the value
```

so we may have a bug somewhere, if this is not satisfied.

how can i reproduce that?